File: lib/models/feedback_note.dart
class FeedbackNote {
  final String id;
  final String sessionId;
  final DateTime date;
  final String topic;
  final int score;
  final List<FeedbackItem> items;
  final List<HintResponseItem> hintResponses;

  FeedbackNote({
    required this.id,
    required this.sessionId,
    required this.date,
    required this.topic,
    required this.score,
    required this.items,
    this.hintResponses = const [],
  });

  factory FeedbackNote.fromJson(Map<String, dynamic> json) {
    final createdAt = json['createdAt'] as String? ?? '';
    return FeedbackNote(
      id: json['id'] as String? ?? '',
      sessionId: json['sessionId'] as String? ?? '',
      date: DateTime.tryParse(createdAt)?.toLocal() ?? DateTime.now(),
      topic: json['topic'] as String? ?? '',
      score: json['score'] as int? ?? 3,
      items: (json['items'] as List<dynamic>? ?? [])
          .map((item) => FeedbackItem.fromJson(item as Map<String, dynamic>))
          .toList(),
      hintResponses: (json['hintResponses'] as List<dynamic>? ?? [])
          .map((item) => HintResponseItem.fromJson(item as Map<String, dynamic>))
          .toList(),
    );
  }
}

class FeedbackItem {
  final String title;
  final String japanese;
  final String pronunciation;
  final String translation;
  final String? description;

  FeedbackItem({
    required this.title,
    required this.japanese,
    required this.pronunciation,
    required this.translation,
    this.description,
  });

  factory FeedbackItem.fromJson(Map<String, dynamic> json) {
    final description = json['description'] as String?;
    return FeedbackItem(
      title: json['title'] as String? ?? '',
      japanese: json['japanese'] as String? ?? '',
      pronunciation: json['pronunciation'] as String? ?? '',
      translation: json['translation'] as String? ?? '',
      description: description != null && description.trim().isEmpty
          ? null
          : description,
    );
  }
}

class HintResponseItem {
  final String userText;
  final String hintText;
  final String hintPronunciation;
  final String hintTranslation;

  HintResponseItem({
    required this.userText,
    required this.hintText,
    this.hintPronunciation = '',
    this.hintTranslation = '',
  });

  factory HintResponseItem.fromJson(Map<String, dynamic> json) {
    return HintResponseItem(
      userText: json['userText'] as String? ?? '',
      hintText: json['hintText'] as String? ?? '',
      hintPronunciation: json['hintPronunciation'] as String? ?? '',
      hintTranslation: json['hintTranslation'] as String? ?? '',
    );
  }
}

File: backend/app/models/feedback_note.py
from pydantic import BaseModel, Field


class FeedbackItemResponse(BaseModel):
    title: str
    japanese: str
    pronunciation: str = ""
    translation: str = ""
    description: str | None = None


class HintResponseItem(BaseModel):
    user_text: str = Field(alias="userText")
    hint_text: str = Field(alias="hintText")
    hint_pronunciation: str = Field(default="", alias="hintPronunciation")
    hint_translation: str = Field(default="", alias="hintTranslation")

    model_config = {"populate_by_name": True, "by_alias": True}


class FeedbackNoteResponse(BaseModel):
    id: str
    session_id: str = Field(alias="sessionId")
    topic: str
    score: int = Field(ge=1, le=5)
    items: list[FeedbackItemResponse]
    hint_responses: list[HintResponseItem] = Field(
        default_factory=list, alias="hintResponses"
    )
    created_at: str = Field(alias="createdAt")

    model_config = {"populate_by_name": True, "by_alias": True}


class FeedbackNoteListResponse(BaseModel):
    notes: list[FeedbackNoteResponse]
